tests/test_event_payload.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2019, Optimizely
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from operator import itemgetter
+
+from optimizely import version
+from optimizely.event.event_payload import Decision, EventBatch, Snapshot, SnapshotEvent, Visitor, VisitorAttribute
+from . import base
+
+
+class EventPayloadTest(base.BaseTest):
+  def _validate_event_object(self, expected_params, event_obj):
```

Comment:
You should ideally override `__eq__` method on the event object and not worry about the logic here.